test/utils/appendHtmlToLink.test.html

```diff
@@ -0,0 +1,206 @@
+<html>
+  <head>
+    <link rel="icon" href="">
+    <link rel="canonical" href="http://localhost:2000/test/utils/appendHtmlToLink.test">
+  </head>
+  <body>
+    <main>
+      <div id="sameserverlinks">
+        <a href="http://localhost:2000/testA"></a>
+        <a href="http://localhost:2000/has/qp?a=1&b=2"></a>
+        <a href="http://localhost:2000/has/hash#blah"></a>
+        <a href="http://localhost:2000/has/qp/and/hash?r=34&hello=world#blah"></a>
+        <a href="http://localhost:2000"></a>
+        <a href="http://localhost:2000/"></a>
+        <a href="http://localhost:2000/endswithslash/"></a>
+        <a href="http://localhost:2000/has/extension.html"></a>
+        <a href="http://localhost:2000/has/json/extension.json"></a>
+        <a href="http://localhost:2000/has/css/extension.css"></a>
+        <a href="http://localhost/wrongport"></a>
+        <a href="http://localhost:2000/#hash"></a>
+        <a href="http://localhost:2000/?q=puppies"></a>
+        <a href="http://localhost:2000/?q=puppies#hash"></a>
+        <a href="http://localhost:2000#hash"></a>
+        <a href="http://localhost:2000?q=puppies"></a>
+        <a href="http://localhost:2000?q=puppies#hash"></a>
+      </div>
+      <div id="relativelinks">
+        <a href="/test/relative/link"></a>
+        <a href="/test/endswith/slash/"></a>
+        <a href="/has/qp?a=1&b=2"></a>
+        <a href="/has/hash#blah"></a>
+        <a href="/has/qp/and/hash?r=34&hello=world#blah"></a>
+        <a href="/test/relative/linkwith.html"></a>
+        <a href="/"></a>
+        <a href=""></a>
+      </div>
+      <div id="externallinks">
+        <a href="http://adobe.com/test"></a>
```

Comment:
adobe.com might be a bad example for an external link, on first sight it looks like an internal one!

Reply:
This was picked up from the initial tests - https://github.com/adobecom/milo/blob/main/test/utils/htmlpostfix.test.html#L35-L37, didn't see a good reason to change it.